src/target.rs

```diff
@@ -78,15 +78,27 @@ impl Target {
             || os == "haiku"
             || os == "illumos"
         {
-            lines.push(format!("-Wl,-soname,lib{}.so.{}", lib_name, major));
+            if capi_config.library.versioning {
+                lines.push(format!("-Wl,-soname,lib{}.so.{}", lib_name, major));
+            } else {
+                lines.push(format!("-Wl,-soname,lib{}.so", lib_name));
+            }
         } else if os == "macos" || os == "ios" {
-            let install_ver = if major == 0 {
-                format!("{}.{}", major, minor)
+            let line = if capi_config.library.versioning {
+                let install_ver = if major == 0 {
+                    format!("{}.{}", major, minor)
+                } else {
+                    format!("{}", major)
+                };
+                format!("-Wl,-install_name,{1}/lib{0}.{5}.dylib,-current_version,{2}.{3}.{4},-compatibility_version,{5}",
+                        lib_name, libdir.display(), major, minor, patch, install_ver)
             } else {
-                format!("{}", major)
+                format!(
+                    "-Wl,-install_name,{1}/lib{0}.dylib",
```

Comment:
@nirbheek Does this match what meson is doing on macOS?

Reply:
Yes, it's close enough to that.

On macOS, Meson allows you to set `current_version` and `compatibility_version` specifically with the [`darwin_versions:` kwarg](https://mesonbuild.com/Reference-manual_functions.html#shared_library). If those are not specified, the default is to use the soname version for both (`soversion:` kwarg). If the `soversion:` kwarg is not specified, the first component of the project version is picked (and this trickles down to everything derived from `soversion`).

`-install_name` is always `libfoo.${soversion}.dylib`, same things as above apply.